sklearn/utils/validation.py

```diff
@@ -100,6 +101,15 @@ def _num_samples(x):
         # Don't get num_samples from an ensembles length!
         raise TypeError('Expected sequence or array-like, got '
                         'estimator %s' % x)
+
+    if isinstance(x, Mapping) and not sp.issparse(x):
+        n_samples = [_num_samples(xx) for xx in x.values()]
+        unique_samples = np.unique(n_samples)
+        if len(unique_samples) > 1:
+                raise ValueError("Inconsistent number of samples in dictionary: %s"
+                                 % (unique_samples))
+        return n_samples[0]
```

Comment:
Since we removed dict support, I am just going to remove this part (doesn't to fail any test by doing that)